reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/Collect.kt

```diff
@@ -7,28 +7,25 @@ import com.badoo.reaktive.single.Single
 import com.badoo.reaktive.single.single
 
 /**
- * Collects elements emitted by the **finite** source [Observable] into a data structure [C]
- * and returns a [Single] that emits this structure. The data structure can be mutable or immutable.
- * The [accumulator] should either mutate the structure and return the same reference,
- * or copy the structure and return a reference to the new copy.
- *
- * Please be aware that the structure may become [frozen](https://github.com/badoo/Reaktive#kotlin-native-pitfalls) in Kotlin/Native.
+ * Collects elements emitted by the **finite** source [Observable] into a mutable data structure [C]
+ * and returns a [Single] that emits this structure. The [accumulator] should mutate the structure
+ * adding elements into it.
  *
  * Please refer to the corresponding RxJava [document](http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#collectInto-U-io.reactivex.functions.BiConsumer-).
  */
-fun <T, C> Observable<T>.collect(initialCollection: C, accumulator: (C, T) -> C): Single<C> =
+fun <T, C> Observable<T>.collect(collectionSupplier: () -> C, accumulator: (C, T) -> Unit): Single<C> =
```

Comment:
Do you want to create parity with RxJava?
`collect(initialCollection: C, ...)`
`collectInto(collectionSupplier: () -> C, ...)`

http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#collectInto-U-io.reactivex.functions.BiConsumer-

Reply:
We can! For me it looks error-prone to specify one instance of a collection. It's easy to miss the other operator and get bugs - every new subscriber will add items to the original collection. WDYT?

Reply:
I got your point, missed it originally. OK, let's keep as you did.